udata/core/badges/models.py

```diff
@@ -14,15 +17,9 @@
 
 
 class Badge(db.EmbeddedDocument):
-    kind = db.StringField(choices=[], required=True)
+    kind = db.StringField(required=True)
     created = db.DateTimeField(default=datetime.now, required=True)
     created_by = db.ReferenceField('User')
-    removed = db.DateTimeField()
-    removed_by = db.ReferenceField('User')
-
-    meta = {
-        'ordering': ['created'],
```

Comment:
Default ordering is only used for document on find without sort.
EmbeddedDocumentList preserve order set on save.